### PR TITLE
Fix skipping some images based on too simple a regex.

### PR DESCRIPTION
**Pull request by matthewwardrop**
Description of changeset: 
Users have been reported (and I have experienced) dropping of various figures from knowledge posts. This is at least part of the problem. I've written a more sophisticated regex expression which now allows more flexible image tags (and does not group them together inadvertently).

Test Plan: 
Run on more complicated markdown scripts to ensure that the images are all collected.

Auto-reviewers: @NiharikaRay @matthewwardrop @earthmancash @danfrankj

### CODE REVIEW

File: knowledge_repo/postprocessors/extract_images.py [hunk 1]
```diff
@@ -50,14 +50,17 @@ def update_thumbnail_uri(cls, kp, images, image_mapping):
     def find_images(cls, md):
         images = []
         images.extend(cls.collect_images_for_pattern(
-            md, '<img.*src=[\'"](.*?)[\'"].*>'))
-        images.extend(cls.collect_images_for_pattern(md, '\!\[.*\]\((.*)\)'))
+            md,
+            r'<img\s+(?:\w+(?:=([\'\"])?(?(1)(?:(?!\1).)*?\1|[^>]*?))?\s+?)*src=([\'\"])(?P<src>(?:(?!\2).)*?)\2(?:\s+\w+(?:=([\'\"])?(?(1)(?:(?!\4).)*?\4|[^>]*?))?)*\s*\/?>'
+        ))
```
Comment (NiharikaRay) [inline knowledge_repo/postprocessors/extract_images.py]: ooooh could we have a comment on what this regexp is supposed to match for future reference? (I'm very bad at reading regexp)

File: knowledge_repo/postprocessors/extract_images.py [hunk 2]
```diff
@@ -50,14 +50,17 @@ def update_thumbnail_uri(cls, kp, images, image_mapping):
...
+        images.extend(cls.collect_images_for_pattern(md, r'\!\[.*\]\((?P<src>[^\)]*)\)'))
+        print(images)
```
Comment (NiharikaRay) [inline knowledge_repo/postprocessors/extract_images.py]: do we still need to print the images?

Comment (matthewwardrop) [inline knowledge_repo/postprocessors/extract_images.py]: Nope. Good catch. Removed.

File: knowledge_repo/postprocessors/extract_images.py (hunk 1 above)
Comment (matthewwardrop) [inline knowledge_repo/postprocessors/extract_images.py]: Added inline comment.

**Pull request merged**